### PR TITLE
feature(@userdive/linker) support area-tag 

### LINKED ISSUE #893: linker plugin should support area tag
# TL;DR

https://github.com/userdive/agent.js/blob/master/packages/linker/src/handler.ts#L7

### PR DESCRIPTION
**Pull request by taka-sho**
# TL;DR

Linker plugin support area tag newly.

Closes #893 

### How to check

*   checkout this pr.
*   yarn install && yarn build
*   yarn run test

### Check list

*   [x] No error

### CODE REVIEW

File: packages/linker/test/handler.test.ts [hunk 1]
```diff
@@ -61,6 +61,22 @@ describe('handler', () => {
     assert(location.href === a.href)
   })
 
+  it('domain of area tag', () => {
+    const l = toLink(internet.url())
+    fixture.set(`
+      <map>
+        <area
+          shape="rect"
+          coords="${random.number},${random.number},${random.number},${random.number}"
```
Comment (shoota) [inline packages/linker/test/handler.test.ts]: fix this. random.number is function.

File: packages/linker/test/handler.test.ts [hunk 2]
```diff
@@ -61,6 +61,22 @@ describe('handler', () => {
     assert(location.href === a.href)
   })
 
+  it('domain of area tag', () => {
```
Comment (shoota) [inline packages/linker/test/handler.test.ts]: what means?

File: packages/linker/src/handler.ts [hunk 3]
```diff
@@ -1,12 +1,14 @@
 export type Domain = string | RegExp
 
+type LinkElements = HTMLAnchorElement | HTMLAreaElement
```
Comment (shoota) [inline packages/linker/src/handler.ts]: badge
I think, it should not need `s`.  This type express **A** 'Element'

File: packages/linker/src/handler.ts [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
+
 export function link (domains: Domain[], linkerParam: string, max: number) {
   return ({ target, srcElement }: Event) => {
     let node = (target || srcElement) as Node
     for (let i = 0; i < max && node; i++) {
-      // TODO need area tag support?
-      if (node instanceof HTMLAnchorElement && linkable(domains, node)) {
-        node.href = linkUrl(node.href, linkerParam)
+      if (linkable(domains, node)) {
+        let href = (node as LinkElements).href
```
Comment (shoota) [inline packages/linker/src/handler.ts]: need this line?

File: packages/linker/src/handler.ts (hunk 3 above)
Comment (taka-sho) [reply packages/linker/src/handler.ts]: i think so too

File: packages/linker/src/handler.ts (hunk 4 above)
Comment (taka-sho) [reply packages/linker/src/handler.ts]: oic

**Comment by shoota** [comment]
Thanks good PR.
Could you add example for area tag?

https://github.com/userdive/agent.js/tree/master/examples/plugin-linker

**Pull request merged**